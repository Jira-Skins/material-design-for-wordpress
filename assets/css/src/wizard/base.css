.material-wizard {
	--mdc-typography-headline5-font-family: "Roboto Mono", monospace;
	--mtb-wizard-primary-color: #212121;
	--mtb-wizard-primary-text: #fff;
	max-width: 1024px;

	& .mdc-button {
		--mdc-theme-primary: var(--mtb-wizard-primary-color);
		--mdc-theme-on-primary: var(--mtb-wizard-primary-text);
	}

	& h3,
	& h4 {
		margin-top: 0;
	}
}

.material_page_material-theme-builder {

	& #adminmenumain,
	& #wpadminbar,
	& #wpfooter,
	& .update-nag {
		display: none;
	}

	& #wpcontent {
		margin-left: 0;
		padding-left: 0;
	}
}

#material-onboarding-wizard {
	--mdc-typography-font-family: "Roboto", sans-serif;
	--mdc-typography-headline3-font-family: "Roboto", sans-serif;
	--mdc-typography-headline4-font-family: "Roboto", sans-serif;
	--mdc-typography-body1-font-family: "Roboto", sans-serif;
	--mdc-typography-button-font-family: "Roboto", sans-serif;
	margin-top: -32px;
}
