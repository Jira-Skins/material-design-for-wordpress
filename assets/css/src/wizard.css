@import "@material/typography/dist/mdc.typography.css";
@import "@material/layout-grid/dist/mdc.layout-grid.css";
@import "@material/button/dist/mdc.button.css";
@import "./wizard/index.css";
