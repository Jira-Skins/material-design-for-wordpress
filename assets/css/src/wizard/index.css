@import "./base.css";
@import "./progress.css";
@import "./navigation.css";
