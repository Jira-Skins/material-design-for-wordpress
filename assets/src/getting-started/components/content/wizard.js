/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import Button from '../../../wizard/components/navigation/button';
import getConfig from '../../get-config';

export const Wizard = props => {
	return (
		<>
			<h2 className="material-gsm__content-title mdc-typography--headline6">
				{ __( 'Onboarding wizard', 'material-theme-builder' ) }
			</h2>
			<p className="material-gsm__content-description">
				Aenean egestas, ante vitae placerat tempor, felis ipsum finibus lectus,
				at eleifend neque tellus ac elit. Praesent cursus lectus felis, a ornare
				metus varius in.
			</p>
			<div className="material-gsm__content-actions">
				<Button
					style="mdc-button--raised"
					text={ __( 'Re-run Wizard', 'material-theme-builder' ) }
					trailingIcon="navigate_next"
					onClick={ props.handleClick }
					link={ getConfig( 'wizardUrl' ) }
				/>
			</div>
		</>
	);
};
