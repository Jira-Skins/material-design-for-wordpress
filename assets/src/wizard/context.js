/**
 * WordPress dependencies
 */
import { createContext, useReducer } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { reducer } from './reducer';
import { STEPS, STATUS, ADDONS } from './constants';

/**
 * Setups context to be used across the app
 */
const StepContext = createContext();

const { Provider } = StepContext;

/**
 * Default state of the world
 *
 */
const initialState = {
	addons: Object.keys( ADDONS ),
	active: STEPS.WELCOME,
	previous: [],
	status: STATUS.IDLE,
	error: {},
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const StepProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default StepContext;
