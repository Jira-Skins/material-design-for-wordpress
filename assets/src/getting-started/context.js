/**
 * Setups context to be used across the app
 *
 */
import { useReducer, createContext } from '@wordpress/element';
import { STATUS } from '../wizard/constants';
import { reducer } from './reducer';
import { TABS } from './constants';

const TabContext = createContext();

const { Provider } = TabContext;

const tabs = Object.keys( TABS );

/**
 * Default state of the world
 *
 */
const initialState = {
	activeTab: 'THEME',
	completed: [ 'WIZARD' ],
	status: STATUS.IDLE,
	actionToInstall: null,
	error: {},
	themeStatus: 'install',
	contentStatus: 'install',
	tabs,
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const TabProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default TabContext;
