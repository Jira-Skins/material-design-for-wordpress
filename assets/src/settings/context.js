/**
 * WordPress dependencies
 */
import { useReducer, createContext } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { STATUS } from '../wizard/constants';
import { reducer } from './reducer';

/**
 * Setups context to be used across the app
 */
const SettingsContext = createContext();

const { Provider } = SettingsContext;

const initialState = {
	status: STATUS.IDLE,
	api: null,
	error: {},
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const SettingsProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default SettingsContext;
