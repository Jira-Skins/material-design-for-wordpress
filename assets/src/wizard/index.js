import React from 'react';
import { render } from '@wordpress/element';
import { StepProvider } from './context';
import ProgressBar from './components/progress-bar';
import Navigation from './components/navigation';
import Content from './components/content';
import Header from './components/header';

const Wizard = () => {
	return (
		<StepProvider>
			<div className="material-wizard mdc-layout-grid">
				<div className="mdc-layout-grid__inner">
					<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-6 mdc-layout-grid__cell--align-middle">
						<Header />
					</div>
					<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-6 mdc-layout-grid__cell--align-middle">
						<ProgressBar />
					</div>

					<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12">
						<Content />
					</div>

					<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12 material-wizard__navigation">
						<Navigation />
					</div>
				</div>
			</div>
		</StepProvider>
	);
};

render( <Wizard />, document.getElementById( 'material-onboarding-wizard' ) );
