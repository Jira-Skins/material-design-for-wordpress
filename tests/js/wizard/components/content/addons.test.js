/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Addons from '../../../../../assets/src/wizard/components/content/addons';
import { StepProvider } from '../../../../../assets/src/wizard/context';

const setup = () => {
	return render(
		<StepProvider>
			<Addons />
		</StepProvider>
	);
};

describe( 'Wizard: Addons', () => {
	beforeAll( () => {
		global.mtbWizard = {
			assetsPath: 'http://example.com/',
			placeholderSmall: 'http://example.com/image-small.png',
			placeholderImage: 'http://example.com/image.png',
		};

		global.mtbOnboarding = {
			themeStatus: 'ok',
		};
	} );

	afterEach( cleanup );

	it( 'matches snapshot', () => {
		const wrapper = setup();
		expect( wrapper ).toMatchSnapshot();
	} );
} );
