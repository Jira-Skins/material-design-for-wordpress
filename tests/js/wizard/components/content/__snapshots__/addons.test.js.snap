// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard: Addons matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-layout-grid__cell--span-12"
      >
        <h3
          class="mdc-typography--headline3 material-wizard__title"
        >
          Install addons
        </h3>
        <div
          class="material-wizard__card mdc-layout-grid__inner mdc-layout-grid__cell--align-middle"
        >
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
          >
            <img
              alt=""
              src="http://example.com/addon-material-theme.png"
            />
          </div>
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-8 mdc-layout-grid__cell--align-middle"
          >
            <h4
              class="mdc-typography--headline4"
            >
              Material Theme
            </h4>
            <p>
              This applies Material Design principles and Material Theming to your site, so you can customize its style.
            </p>
          </div>
        </div>
        <hr />
        <div
          class="material-wizard__card mdc-layout-grid__inner mdc-layout-grid__cell--align-middle"
        >
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
          >
            <img
              alt=""
              src="http://example.com/addon-quick-start-examples.png"
            />
          </div>
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-8 mdc-layout-grid__cell--align-middle"
          >
            <h4
              class="mdc-typography--headline4"
            >
              Quick Start Examples
            </h4>
            <p>
              Contains a number of page examples for home, about, a projects page, blog, and a contact page. These layouts show different ways you can use Material Components to address common user needs.
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-layout-grid__cell--span-12"
    >
      <h3
        class="mdc-typography--headline3 material-wizard__title"
      >
        Install addons
      </h3>
      <div
        class="material-wizard__card mdc-layout-grid__inner mdc-layout-grid__cell--align-middle"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
        >
          <img
            alt=""
            src="http://example.com/addon-material-theme.png"
          />
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-8 mdc-layout-grid__cell--align-middle"
        >
          <h4
            class="mdc-typography--headline4"
          >
            Material Theme
          </h4>
          <p>
            This applies Material Design principles and Material Theming to your site, so you can customize its style.
          </p>
        </div>
      </div>
      <hr />
      <div
        class="material-wizard__card mdc-layout-grid__inner mdc-layout-grid__cell--align-middle"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
        >
          <img
            alt=""
            src="http://example.com/addon-quick-start-examples.png"
          />
        </div>
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-8 mdc-layout-grid__cell--align-middle"
        >
          <h4
            class="mdc-typography--headline4"
          >
            Quick Start Examples
          </h4>
          <p>
            Contains a number of page examples for home, about, a projects page, blog, and a contact page. These layouts show different ways you can use Material Components to address common user needs.
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
